### ISSUE TITLE
When backend is run for the first time, a database error occurs

### ISSUE THREAD

**Issue by bartekpacia** [kacaleksandra/notes#14]
Reproduction:

1. Kill all running containers
2. `docker compose up --build` and observe for a minute or two

After some time, the following error appears:

```
postgres-1  | 2024-01-30 22:37:00.021 UTC [34] ERROR:  relation "public.Reminders" does not exist at character 130
postgres-1  | 2024-01-30 22:37:00.021 UTC [34] STATEMENT:  SELECT "public"."Reminders"."id", "public"."Reminders"."date", "public"."Reminders"."userId", "public"."Reminders"."noteId" FROM "public"."Reminders" WHERE ("public"."Reminders"."date" >= $1 AND "public"."Reminders"."date" <= $2) OFFSET $3
app-1       | [Nest] 7  - 01/30/2024, 10:37:00 PM   ERROR [Scheduler] PrismaClientKnownRequestError:
app-1       | Invalid `this.prisma.reminders.findMany()` invocation in
app-1       | /dist/src/notifications/notifications.service.js:79:55
app-1       |
app-1       |   76 async checkAndSendNotifications() {
app-1       |   77     const currentDateTime = new Date();
app-1       |   78     const warsawTime = currentDateTime.getTime() + 60 * 60 * 1000;
app-1       | → 79     const reminders = await this.prisma.reminders.findMany(
app-1       | The table `public.Reminders` does not exist in the current database
```

You can notice that it occurs every minute (every time a notification scheduler tries to send notifications):

<details><summary>Full logs</summary>

```
$ docker compose up --build                                                                                                       exit 130
[+] Building 0.2s (19/19) FINISHED                                                                                    docker:desktop-linux
 => [app internal] load build definition from Dockerfile                                                                              0.0s
 => => transferring dockerfile: 831B                                                                                                  0.0s
 => [app internal] load .dockerignore                                                                                                 0.0s
 => => transferring context: 374B                                                                                                     0.0s
 => [app internal] load metadata for docker.io/library/node:20                                                                        0.2s
 => [app builder 1/7] FROM docker.io/library/node:20@sha256:ffebb4405810c92d267a764b21975fb2d96772e41877248a37bf3abaa0d3b590          0.0s
 => [app internal] load build context                                                                                                 0.0s
 => => transferring context: 4.65kB                                                                                                   0.0s
 => CACHED [app stage-1 2/8] RUN apt-get update                                                                                       0.0s
 => CACHED [app stage-1 3/8] RUN apt-get install dumb-init                                                                            0.0s
 => CACHED [app builder 2/7] WORKDIR /app                                                                                             0.0s
 => CACHED [app builder 3/7] COPY package*.json ./                                                                                    0.0s
 => CACHED [app builder 4/7] COPY prisma ./prisma/                                                                                    0.0s
 => CACHED [app builder 5/7] RUN npm install                                                                                          0.0s
 => CACHED [app builder 6/7] COPY . .                                                                                                 0.0s
 => CACHED [app builder 7/7] RUN npm run build                                                                                        0.0s
 => CACHED [app stage-1 4/8] COPY --from=builder /app/node_modules ./node_modules                                                     0.0s
 => CACHED [app stage-1 5/8] COPY --from=builder /app/package*.json ./                                                                0.0s
 => CACHED [app stage-1 6/8] COPY --from=builder /app/dist ./dist                                                                     0.0s
 => CACHED [app stage-1 7/8] COPY --from=builder /app/tsconfig.json ./                                                                0.0s
 => CACHED [app stage-1 8/8] COPY --from=builder /app/prisma ./prisma                                                                 0.0s
 => [app] exporting to image                                                                                                          0.0s
 => => exporting layers                                                                                                               0.0s
 => => writing image sha256:5ed0293064d91f1d7344ed38e58ba6bc32a9b84e254f3092ee1442c98520ed4b                                          0.0s
 => => naming to docker.io/library/backend-app                                                                                        0.0s
[+] Running 2/0
 ✔ Container backend-postgres-1  Created                                                                                              0.0s
 ✔ Container backend-app-1       Created                                                                                              0.0s
Attaching to app-1, postgres-1
postgres-1  |
postgres-1  | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres-1  |
postgres-1  | 2024-01-30 22:32:23.341 UTC [1] LOG:  starting PostgreSQL 16.1 (Debian 16.1-1.pgdg120+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
postgres-1  | 2024-01-30 22:32:23.342 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres-1  | 2024-01-30 22:32:23.342 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres-1  | 2024-01-30 22:32:23.343 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres-1  | 2024-01-30 22:32:23.347 UTC [29] LOG:  database system was shut down at 2024-01-30 22:32:21 UTC
postgres-1  | 2024-01-30 22:32:23.349 UTC [1] LOG:  database system is ready to accept connections
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [NestFactory] Starting Nest application...
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [InstanceLoader] AppModule dependencies initialized +12ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [InstanceLoader] PassportModule dependencies initialized +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [InstanceLoader] JwtModule dependencies initialized +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [InstanceLoader] DiscoveryModule dependencies initialized +1ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [InstanceLoader] ScheduleAppModule dependencies initialized +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [InstanceLoader] ScheduleModule dependencies initialized +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [InstanceLoader] AuthModule dependencies initialized +1ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [InstanceLoader] UsersModule dependencies initialized +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [InstanceLoader] CategoriesModule dependencies initialized +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [InstanceLoader] NotesModule dependencies initialized +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [InstanceLoader] NotificationsModule dependencies initialized +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [InstanceLoader] TokenModule dependencies initialized +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RoutesResolver] AuthController {/api/auth}: +15ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/auth/login, POST} route +1ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RoutesResolver] UsersController {/api/users}: +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/users, POST} route +1ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/users, GET} route +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/users/:id, GET} route +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RoutesResolver] CategoriesController {/api/categories}: +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/categories, POST} route +1ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/categories/:id, DELETE} route +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/categories, GET} route +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/categories/:id, GET} route +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RoutesResolver] NotesController {/api/notes}: +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/notes, POST} route +1ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/notes/:id, PATCH} route +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/notes/:id, DELETE} route +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/notes, GET} route +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/notes/:id, GET} route +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/notes/by-category/:categoryId, GET} route +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RoutesResolver] NotificationsController {/api/notifications}: +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/notifications, POST} route +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/notifications/:reminderId, DELETE} route +1ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RoutesResolver] TokenController {/api/token}: +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/token, POST} route +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [RouterExplorer] Mapped {/api/token/:tokenId, DELETE} route +0ms
app-1       | [Nest] 7  - 01/30/2024, 10:32:23 PM     LOG [NestApplication] Nest application successfully started +9ms
app-1       | Running every minute
postgres-1  | 2024-01-30 22:33:00.083 UTC [34] ERROR:  relation "public.Reminders" does not exist at character 130
postgres-1  | 2024-01-30 22:33:00.083 UTC [34] STATEMENT:  SELECT "public"."Reminders"."id", "public"."Reminders"."date", "public"."Reminders"."userId", "public"."Reminders"."noteId" FROM "public"."Reminders" WHERE ("public"."Reminders"."date" >= $1 AND "public"."Reminders"."date" <= $2) OFFSET $3
app-1       | [Nest] 7  - 01/30/2024, 10:33:00 PM   ERROR [Scheduler] PrismaClientKnownRequestError:
app-1       | Invalid `this.prisma.reminders.findMany()` invocation in
app-1       | /dist/src/notifications/notifications.service.js:79:55
app-1       |
app-1       |   76 async checkAndSendNotifications() {
app-1       |   77     const currentDateTime = new Date();
app-1       |   78     const warsawTime = currentDateTime.getTime() + 60 * 60 * 1000;
app-1       | → 79     const reminders = await this.prisma.reminders.findMany(
app-1       | The table `public.Reminders` does not exist in the current database.
app-1       | Running every minute
postgres-1  | 2024-01-30 22:34:00.013 UTC [34] ERROR:  relation "public.Reminders" does not exist at character 130
postgres-1  | 2024-01-30 22:34:00.013 UTC [34] STATEMENT:  SELECT "public"."Reminders"."id", "public"."Reminders"."date", "public"."Reminders"."userId", "public"."Reminders"."noteId" FROM "public"."Reminders" WHERE ("public"."Reminders"."date" >= $1 AND "public"."Reminders"."date" <= $2) OFFSET $3
app-1       | [Nest] 7  - 01/30/2024, 10:34:00 PM   ERROR [Scheduler] PrismaClientKnownRequestError:
app-1       | Invalid `this.prisma.reminders.findMany()` invocation in
app-1       | /dist/src/notifications/notifications.service.js:79:55
app-1       |
app-1       |   76 async checkAndSendNotifications() {
app-1       |   77     const currentDateTime = new Date();
app-1       |   78     const warsawTime = currentDateTime.getTime() + 60 * 60 * 1000;
app-1       | → 79     const reminders = await this.prisma.reminders.findMany(
app-1       | The table `public.Reminders` does not exist in the current database.
app-1       | Running every minute
postgres-1  | 2024-01-30 22:35:00.020 UTC [34] ERROR:  relation "public.Reminders" does not exist at character 130
postgres-1  | 2024-01-30 22:35:00.020 UTC [34] STATEMENT:  SELECT "public"."Reminders"."id", "public"."Reminders"."date", "public"."Reminders"."userId", "public"."Reminders"."noteId" FROM "public"."Reminders" WHERE ("public"."Reminders"."date" >= $1 AND "public"."Reminders"."date" <= $2) OFFSET $3
app-1       | [Nest] 7  - 01/30/2024, 10:35:00 PM   ERROR [Scheduler] PrismaClientKnownRequestError:
app-1       | Invalid `this.prisma.reminders.findMany()` invocation in
app-1       | /dist/src/notifications/notifications.service.js:79:55
app-1       |
app-1       |   76 async checkAndSendNotifications() {
app-1       |   77     const currentDateTime = new Date();
app-1       |   78     const warsawTime = currentDateTime.getTime() + 60 * 60 * 1000;
app-1       | → 79     const reminders = await this.prisma.reminders.findMany(
app-1       | The table `public.Reminders` does not exist in the current database.
```

</details>

**Comment by P3T3R3** [comment]
Please let me know if this hotfix fixes the issue

**Comment by bartekpacia** [comment]
Yes, issues are fixed. 🎉 

I'll create another issue for adding a simple smoke test/integration test to CI.

**Comment by bartekpacia** [comment]
fixed by #15
